packages/linux-firmware/linux-firmware.spec

```diff
@@ -0,0 +1,89 @@
+%global debug_package %{nil}
+
+%global _firmwarepath   /usr/lib/firmware
+
+# Many of the firmware files have specialized binary formats that are not supported
+# by the strip binary used in __spec_install_post macro. Work around build failures
+# by skipping striping.
+%global __strip /usr/bin/true
+
+Name:       %{_cross_os}linux-firmware
+Version:    20230625
+Release:    1%{?dist}
+Summary:    Firmware files used by the Linux kernel
+License:    GPL+ and GPLv2+ and MIT and Redistributable, no modification permitted
+URL:        https://www.kernel.org/
+
+Source0:    https://www.kernel.org/pub/linux/kernel/firmware/linux-firmware-%{version}.tar.xz
```

Comment:
That is a good question. There should not be a dependency of a certain firmware package state requiring a certain kernel version. The worst that could happen there would be additional firmware files the kernel does not know about. Or installing compressed firmware the kernel does not know how to decompress (See comment in spec file).

The other way around we may get into situations where a driver expects certain firmware files which are not there in which case we need to update linux-firmware.

If we want/need a hard dependency between firmware packages is debatable I guess. Maybe make it a rule to just update it in lockstep with major kernel updates is an option?

Reply:
It would be drivers checking for firmware versions and features, so using newer firmware with older drivers would work. In other cases, drivers would explicitly request a particular firmware version to be loaded (and older firmware files be kept around in new releases). Leo's suggestion of updating with the introduction of a new kernel series at the latest seems sound to me, but more frequently would be fine, too.